lib/Fatal.pm

```diff
@@ -48,7 +53,9 @@ use constant ERROR_AUTODIE_CONFLICT => q{"no autodie '%s'" is not allowed while
 
 use constant ERROR_FATAL_CONFLICT => q{"use Fatal '%s'" is not allowed while "no autodie '%s'" is in effect};
 
-use constant ERROR_58_HINTS => q{Non-subroutine %s hints for %s are not supported under Perl 5.8.x};
+use constant ERROR_SMARTMATCH_HINTS => q{%s hints for %s must be code, regexp, or undef under Perl 5.8 or 5.42 or newer};
+
+use constant WARNING_SMARTMATCH_DEPRECATED => q{Hints other than code, regexp, or undef are deprecated and will be removed in Perl 5.42};
```

Comment:
I would not say "removed in Perl 5.42" I would say "removed by Perl 5.42". In practice we will remove support for smartmatch sometime in 5.41.

Reply:
Not sure if the emphasis is clear here. Replace "in" with "by".